packages/shared/invokeGuardedCallbackImpl.js

```diff
@@ -59,24 +58,9 @@ if (__DEV__) {
     const fakeNode = document.createElement('react');
 
     invokeGuardedCallbackImpl = function invokeGuardedCallbackDev<
-      A,
-      B,
-      C,
-      D,
-      E,
-      F,
+      Args: Array<mixed>,
       Context,
-    >(
-      name: string | null,
-      func: (a: A, b: B, c: C, d: D, e: E, f: F) => mixed,
-      context: Context,
-      a: A,
-      b: B,
-      c: C,
-      d: D,
-      e: E,
-      f: F,
-    ) {
+    >(name: string | null, func: (...Args) => mixed, context: Context): void {
```

Comment:
This only works because we're not calling it with anything more specific, right? Othewise it might be an error to use this now because it would just pass "mixed" to the callback.

Reply:
As far as I can tell, this works correctly:

https://flow.org/try/#0CYUwxgNghgTiAEAzArgOzAFwJYHtXwHcYoAHAHgEEYBzAZwC54riBPMgWywA8RgA+PgApBAOjFU6ASngBePvE49gkxoIDkqEAXixqagDTwxIibWlz4ANxxZgAbgBQD0JFgIU6bHiSCojVMjsAEYgMIZBjLQYMFio1CpWNvZOYHhRhMQkJLyyGaSCiJKODkSk2cDqmtq6BvAAjIZqtGpFJZnllVo6NLVNvc2tQA